### PR TITLE
Wip/hughsie/spi

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/ch341a/fu-ch341a-cfi-device.c [hunk 1]
```diff
@@ -106,14 +106,17 @@ fu_ch341a_cfi_device_read_jedec(FuCh341aCfiDevice *self, GError **error)
 		g_set_error_literal(error,
 				    FWUPD_ERROR,
 				    FWUPD_ERROR_NOT_SUPPORTED,
-				    "flash ID non-valid");
+				    "flash ID non-valid, got 0x000000");
```
Comment (superm1) [inline plugins/ch341a/fu-ch341a-cfi-device.c:109]: maybe flash-ID is empty?

Comment (hughsie) [reply plugins/ch341a/fu-ch341a-cfi-device.c:109]: It's really for the case where you forgot to connect a wire, e.g. CS

File: plugins/ch341a/fu-ch341a-cfi-device.c [hunk 2]
```diff
@@ -106,14 +106,17 @@ fu_ch341a_cfi_device_read_jedec(FuCh341aCfiDevice *self, GError **error)
...
 	}
 	if (buf[1] == 0xFF && buf[2] == 0xFF && buf[3] == 0xFF) {
-		g_set_error_literal(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_NOT_SUPPORTED,
-				    "device not detected");
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "device not detected, flash ID 0x%02X%02X%02X",
```
Comment (superm1) [inline plugins/ch341a/fu-ch341a-cfi-device.c:116]: flash ID not programmed?

**Review by superm1 — APPROVED** [review]
I'm not sure about the specific error strings; but otherwise fine.

File: plugins/ch341a/fu-ch341a-cfi-device.c (hunk 2 above)
Comment (hughsie) [reply plugins/ch341a/fu-ch341a-cfi-device.c:116]: Not usually; 99% of the time this is just a missing entry in cfi.quirk

**Pull request merged**